### DISCUSSION TITLE
changing auth email templates - self-hosted with docker

### DISCUSSION THREAD

**Discussion by damien-schneider** [supabase/supabase#17342] [Questions] [answered]
I have self-hosted supabase with docker, everything is working fine, but I'm getting trouble with the auth.

I cannot find a way to change the default email template
![image](https://github.com/supabase/supabase/assets/74979845/421ac5e0-9744-4a35-a6ea-90177d6d4ad8)

I have tried to setup the config.toml with the templates directory and the different html files,
but I'm not even sure where to put them, even if I think I did it well, it doesn't work as well...
https://supabase.com/docs/guides/cli/customizing-email-templates

The problem is not only the language and the style that is default but the link provided in the confirmation mail is the following one (which as you can see, doesn't use my SITE_URL env variable

```http://127.0.0.1/auth/v1/verify?token=**TOKEN HERE**&type=signup&redirect_to=https://my-url```

If someone can help me; I would love to contribute to the doc after to clarify this point...

Feel free to ask more details if needed, thanks by advance !

**Comment by silentworks** [comment]
Can you clarify if you are using the CLI development setup vs self-hosted docker-compose setup? this is generally where the confusion comes in for many users, the docs that you link to is only if you are using `supabase start` from the supabase CLI.

**Reply by damien-schneider** [reply]
Ok sorry for the confusion and thanks for the reply,
I'm actually using self-hosted docker-compose on a VPS

**Reply by silentworks** [reply, marked as answer]
We need to document this in the appropriate section of the docs as it seems even the correct section is still referencing the `config.toml` which it shouldn't.

This is the correct section of the docs https://supabase.com/docs/guides/self-hosting/auth/config, but as I stated above the information here is incorrect.

You will need to add environment variables to the `auth` section of the `docker-compose.yml` file. The missing variable can be found here https://github.com/supabase/gotrue#e-mail, an example of one is `MAILER_TEMPLATES_CONFIRMATION` which you would add the `GOTRUE_` prefix to, so it would be `GOTRUE_MAILER_TEMPLATES_CONFIRMATION` in your `docker-compose.yml` file. Do note that these need to be http URLs as you will need to host these templates with a webserver.

**Reply by damien-schneider** [reply]
Thank you very much for taking of your time to answer, I will try everything you mentionned, I keep you in touch.

Thanks a lot !

**Reply by damien-schneider** [reply]
With the url modified, it is actually not returning the default mail (which is a first great step, thanks !).
But it returns now `You need to enable JavaScript to run this app.`

![image](https://github.com/supabase/supabase/assets/74979845/626f7cf1-00f4-45bd-a6b2-17dffdba8312)

Again, I've tried many things before asking to be sure to not take your precious time for nothing, but impossible to make it work

The URL seems to return a simple html without javascript as show the curl request just below :
(I have also removed the href, to be sure it doesn't come from it (even if it was obviously not the case ^^)

```StatusCode        : 200
StatusDescription : OK
Content           : <!DOCTYPE html>
                    <html>
                    <body>
                      <h2>SoFI</h2>
                      <h2>Veuillez confirmer votre inscription</h2>

                      <p>Suivez ce lien pour confirmer votre inscription :</p>
                    </body>
                    </html>

RawContent        : HTTP/1.1 200 OK
                    Connection: keep-alive
                    Accept-Ranges: bytes
                    Content-Length: 170
                    Content-Type: text/html
                    Date: Wed, 13 Sep 2023 20:22:31 GMT
                    ETag: "650219dc-aa"
                    Last-Modified: Wed, 13 Sep 2023 2...
Forms             : {}
Headers           : {[Connection, keep-alive], [Accept-Ranges, bytes], [Content-Length, 170], [Content-Type,
                    text/html]...}
Images            : {}
InputFields       : {}
Links             : {}
ParsedHtml        : mshtml.HTMLDocumentClass
RawContentLength  : 170
```

Sorry again for taking your time, but thanks so much by advance !

**Reply by silentworks** [reply]
That's very weird that your email looks like that, maybe try sending it to a different email provider as this might be a Gmail issue.

**Reply by damien-schneider** [reply]
Thanks for your reply, and sorry for my late one. I finally resolved my problem, I was wrongly serving my html because it is needed to serve by a `https://domain.xx/email.html` the `.html` is necessary for the endpoint.

All is working thanks to your first reply, and the good documentation redirect, thanks again !!
I will try to contribute by proposing an update for the documentation.
